src/main/java/com/nha/abdm/wrapper/hip/facade/link/HIPFacadeLinkController.java

```diff
@@ -67,4 +76,24 @@ public FacadeResponse verifyOtp(@RequestBody VerifyOTP verifyOTP) {
   public FacadeResponse addPatients(@RequestBody List<Patient> patients) {
     return workflowManager.addPatients(patients);
   }
+
+  /**
+   * Convert JsonProcessingException exceptions thrown by HIP Facade Link controller to API error
+   * response.
+   */
+  @ExceptionHandler(JsonProcessingException.class)
+  private FacadeResponse handleJsonProcessingException(JsonProcessingException ex) {
+    return FacadeResponse.builder()
+        .message(ex.getMessage())
+        .code(HttpStatus.BAD_REQUEST.value())
+        .build();
+  }
+
+  @ExceptionHandler(IllegalDataStateException.class)
+  private FacadeResponse handleIllegalDataStateException(Exception ex) {
+    return FacadeResponse.builder()
+        .message(ex.getMessage())
+        .code(HttpStatus.INTERNAL_SERVER_ERROR.value())
```

Comment:
This looks like an invalid request from the client (since he has sent invalid requestId), so we can probably send `HttpStatus.BAD_REQUEST` instead.

Reply:
request Id is one scenario. at any point if the database does not have required state, we can use this exception. client can still send a valid request id and for some reason if for that request id, there were no records stored then this exception works best in that scenario.